### PR TITLE
Add other cases for dynamically fetching $ref schemas

### LINKED ISSUE #56: Add other cases for dynamically fetching $ref schemas
On expand, $ref type schemas should dynamically fetch the schemas. The following are the common ways a $ref's source may be defined as:

1. References a __sub-schema__ within the current schema ( ex. `schemas/refType/simpleReference.json` ) -> ( resolved in #54 ✔️  )
```json
$ref: '#definitions/name'
```

2. References a __URI__ (ex. `schemas/demo/metadata-metaschema.json`)
```json
$ref: 'http://json-schema.org/draft-06/schema#'
```

3. References a __file using the current path__ (ex. `schemas/demo/worker-list.json`)
```json
$ref: 'worker-full.json#'
```

### PR DESCRIPTION
**Pull request by JiwoonKim**
Closes #56 

**Applied Changes**
- [x] pass list of references to use for dereferencing
- [x] create map of references: `$id: schema`
- [x] add `$id` properties to all schema examples
- [x] enable fetching based on reference map
   - [x] `worker-full.json#` -> find schema with $id of `/schemas/worker-manager/v1/worker-full.json#` in references map
   - [x] if ref schema does not exist in references map, pass error message to be displayed in table
- [x] update storybook examples to use references

### CODE REVIEW

**Comment by helfi92** [comment]
Screen Shot 2020-02-18 at 10 31 11 AM

**Comment by helfi92** [comment]
Accidentally approved this PR. Please flag me for a review once this is ready for review.

File: src/utils/schemaTree.js [hunk 1]
```diff
@@ -22,6 +22,9 @@ import { COMBINATION_TYPES, REF_TYPE, CUSTOM_KEYWORDS } from './constants';
  *                         (a node object as illustrated above)
  *   isExpanded: ..     // whether the node is expanded or not
  * }
+ *
+ * @param {object} schema
+ * @param {array} path path to current node/subtree
```
Comment (helfi92) [inline src/utils/schemaTree.js]: Could you put an example of a path here, please? :-)

File: src/components/NormalLeftRow/index.jsx [hunk 2]
```diff
@@ -262,10 +269,17 @@ NormalLeftRow.propTypes = {
    * Only necessary for ref rows.
    */
   setSchemaTree: func,
+  /**
+   * Object where all schemas are stored so that the table
+   * can reference to for dereferencing $ref schemas.
+   * Only necessary for ref rows.
+   */
```
Comment (helfi92) [inline src/components/NormalLeftRow/index.jsx]: What should the object key be? A comment here would be helpful.

File: src/components/SchemaViewer/index.jsx [hunk 3]
```diff
@@ -35,7 +44,11 @@ function SchemaViewer({ schema }) {
       {sourceMode ? (
         <SourceView schema={schema} />
       ) : (
-        <SchemaTable schemaTree={schemaTree} setSchemaTree={setSchemaTree} />
+        <SchemaTable
+          schemaTree={schemaTree}
+          setSchemaTree={setSchemaTree}
+          references={referenceMap}
```
Comment (helfi92) [inline src/components/SchemaViewer/index.jsx]: Do we have an example in the storybook where the schema is not able to find the reference? I think that would be good to have.

File: src/components/NormalLeftRow/index.jsx (hunk 2 above)
Comment (JiwoonKim) [reply src/components/NormalLeftRow/index.jsx]: Applied changes in #101

File: src/utils/schemaTree.js (hunk 1 above)
Comment (JiwoonKim) [reply src/utils/schemaTree.js]: Applied changes in #101 ! (Also further examples of path are also included in the comment above :)

**Pull request merged**